src/main/java/seedu/jarvis/model/financetracker/FinanceTracker.java

```diff
@@ -132,7 +99,7 @@ public void addSinglePurchase(Purchase purchase) {
      *
      * @param itemNumber of payment to be deleted
      */
-    public Purchase deleteSinglePurchase(int itemNumber) {
+    public Purchase deleteSinglePurchase(int itemNumber) throws PurchaseNotFoundException {
```

Comment:
Runtime exception no need to declare in method signature, but I am not too sure on which is better. There is an argument that it may be confusing to differentiate between checked exceptions, and can also clutter logic. But I am not too sure how valid these opinions are.

Reply:
alright thanks for the tip!